src/main/java/com/twilio/oai/TwilioJavaGenerator.java

```diff
@@ -237,6 +250,9 @@ private void flattenStringMap(final Map<String, Object> resource, final String k
     }
 
     private String getResourceName(final String path) {
+        if (inflector.isAbbrevation(PathUtils.getLastPathPart(PathUtils.cleanPath(path)))) {
```

Comment:
nit: should save off the result to `cleanPath` and `getLastPathPart` so they don't need to be called multiple times.